crates/bifrost/src/providers/replicated_loglet/record_cache.rs

```diff
@@ -22,4 +35,63 @@ impl RecordCache {
     pub fn new(_memory_budget_bytes: usize) -> Self {
         Self {}
     }
+
+    /// Append the next batch to the loglet_id cache.
+    ///
+    /// Cache expecting offsets to come in order. If the provided
+    /// `first_offset` would create a gap in the cache the append
+    /// will be rejected with an error.
+    pub(crate) fn append_records(
+        &self,
+        _loglet_id: ReplicatedLogletId,
+        _first_offset: LogletOffset,
+        _records: Arc<[Record]>,
+    ) -> Result<(), RecordCacheError> {
+        todo!()
+    }
+
+    /// Release offset (inclusive) to readers.
+    ///
+    /// All active reader streams should receive all released batches.
+    pub(crate) fn release_records(&self, _loglet_id: ReplicatedLogletId, _offset: LogletOffset) {
```

Comment:
I am considering dropping this. And make sure that caller of append_records must only do so for committed batches. This way Appenders (in case of local sequencer) will just append their batch to the cache before notifying change to the known global committed offset

Reply:
That makes sense.